### PR TITLE
Add option to skaffold run similar to "skaffold build -b"

### PR DESCRIPTION
**Pull request by anshlykov**
Fixes: #4494

**Description**
Added the artifact filter for the `run` command, similar to the `build` command.

### CODE REVIEW

**Comment by anshlykov** [comment]
@googlebot I signed it!

File: cmd/skaffold/app/cmd/run.go [hunk 1]
```diff
@@ -36,13 +37,16 @@ func NewCmdRun() *cobra.Command {
 		WithExample("Build, test, deploy and tail the logs", "run --tail").
 		WithExample("Run with a given profile", "run -p <profile>").
 		WithCommonFlags().
+		WithFlags(func(f *pflag.FlagSet) {
+			f.StringSliceVarP(&opts.TargetImages, "build-image", "b", nil, "Choose which artifacts to build. Artifacts with image names that contain the expression will be built only. Default is to build sources for all artifacts")
```
Comment (gsquared94) [inline cmd/skaffold/app/cmd/run.go:41]:
```suggestion
			f.StringSliceVarP(&opts.TargetImages, "build-image", "b", nil, "Only build artifacts with image names that contain the given substring. Default is to build sources for all artifacts")
```
nit: I know that you took the same text as the existing `build-image` flag on `build`. But the grammar seems off.

**Review by gsquared94 — APPROVED** [review]
LGTM

File: cmd/skaffold/app/cmd/run.go (hunk 1 above)
Comment (briandealwis) [reply cmd/skaffold/app/cmd/run.go:41]: If the flag is the same elsewhere then we should instead add it to the [`flagRegistry`](https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/flags.go#L53).

File: cmd/skaffold/app/cmd/run_test.go [hunk 2]
```diff
@@ -37,3 +43,60 @@ func TestNewCmdRun(t *testing.T) {
...
+							},
+							{
+								ImageName: "second-test",
+							},
+							{
+								ImageName: "test",
+							},
+							{
+								ImageName: "aaabbbccc",
+							},
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/run_test.go:88]: These take up a lot of space!

```suggestion
							{ImageName: "first"},
							{ImageName: "second-test"},
							{ImageName: "test"},
							{ImageName: "aaabbbccc"},
```

**Comment by gsquared94** [comment]
@anshlykov thanks for making all the suggested changes. Can you sync with the master branch; then we can merge this and call it done 👍

**Comment by gsquared94** [comment]
Travis bug is showing duplicate checks and stuck on one of them. This PR's actually cleared all checks. Merging.

**Comment by anshlykov** [comment]
@gsquared94 thanks for resolving the merge conflict

**Pull request merged**